irdb/tests/test_publish.py

```diff
@@ -1,104 +1,235 @@
-
-
-def test_run_main(preserve_versions):
-    argv = ["", "-c", "test_package", "-p", "<insert_if_running_locally"]
-    pub.main(argv)
-    with open(PATH_TEST_PACKAGE_VERSION_YAML, encoding="utf8") as f:
-        version_dict = yaml.safe_load(f)
-    assert version_dict["version"] == NOW
-
-
-def test_run_main_no_update_version():
-    argv = ["", "-c", "test_package", "-p", "<insert_if_running_locally", "-d"]
-    pub.main(argv)
-
-    with open(PATH_TEST_PACKAGE_VERSION_YAML, encoding="utf8") as f:
-        version_dict = yaml.safe_load(f)
-    assert version_dict["version"] == "2022-07-11.dev"
-
-    with open(pub.PATH_PACKAGES_YAML, encoding="utf8") as f:
-        pckgs_dict = yaml.safe_load(f)
-    assert pckgs_dict["test_package"]["stable"] == "test_package.2022-07-11"
-
-
-# def test_versions():
-#     """See whether the versions are compatible."""
-# 
-#     release_from_release = {
-#         "stable": "stable",
-#         "dev": "latest",
-#         "latest": "dev",
-#     }
-# 
-#     with open(pub.PATH_PACKAGES_YAML, encoding="utf8") as f:
-#         pckgs_dict = yaml.safe_load(f)
-#
-#     for name, package_dict in pckgs_dict.items():
-#         path_version = pub.PKGS_DIR / name / "version.yaml"
-#         if not path_version.exists():
-#             # TODO: should this be disallowed?
-#             continue
-#         with open(path_version, encoding="utf8") as f:
-#             version_dict = yaml.safe_load(f)
-#             release_1 = version_dict["release"]
-#             version_1 = f"{name}.{version_dict['version']}"
-#             release_2 = release_from_release[release_1]
-#             version_2 = package_dict[release_2]
-#             assert version_1 == version_2
-
-
-# def rename_zips():
-#     from datetime import datetime as dt
-#     from glob import glob
-#     import os
-#
-#     os.chdir("F:/Work/irdb/_OLD_FILES")
-#     for fname in glob("*.zip"):
-#         base = fname.split(".")[0]
-#         mod_date = str(dt.fromtimestamp(os.path.getmtime(fname)))[:10]
-#         new_name = f"{base}.{mod_date}.zip"
-#         print(fname, new_name)
-#         os.rename(fname, new_name)
+    current_stable = "2023-07-10"
+    prompt = ("This will supersede the current STABLE version "
+              f"({current_stable}) of '{pkg_name}' on the IRDB server. "
+              "The uploaded package will be set as the new default for "
+              f"'{pkg_name}'.\nAre you sure you want to continue?    (y)/n:  ")
+    with mock.patch.object(builtins, "input",
+                           partial(_fake_input, from_user=from_user)):
+        with mock.patch("sys.stdout", new_callable=StringIO) as mock_stdout:
+            assert result == pub.confirm(pkg_name)
+            assert prompt == mock_stdout.getvalue().strip("\n")
+
+
+@pytest.fixture(name="default_argv", scope="function")
+def fixture_default_argv():
+    return ["", "-l", "fake_username", "-p", "fake_password", "test_package"]
+
+
+@pytest.mark.xfail
```

Comment:
This test fails because there is no `test_package.2023-07-20.zip` and no `test_package.2023-07-20.dev.zip` in `_ZIPPED_PACKAGES`.

I´d be fine if we just commit those two files to the repository. Or we could create them in the test, or in a fixture, or we just keep the xfail, all fine with me.

Reply:
That's what I attempted to do with the `temp_zipfiles` fixture. Obviously didn't work for some reason. It did work on my machine, I also don't have these two versions of the `test_package` in `_ZIPPED_PACKAGES`. I think the way to go would be to try and make the fixture approach work (at a later point).